src/platform/plugins/shared/visualizations/public/embeddable/visualize_embeddable.tsx

```diff
@@ -306,7 +307,18 @@ export const getVisualizeEmbeddableFactory: (deps: {
       },
       canLinkToLibrary: () => Promise.resolve(!linkedToLibrary),
       canUnlinkFromLibrary: () => Promise.resolve(linkedToLibrary),
-      checkForDuplicateTitle: () => Promise.resolve(), // Handled by saveToLibrary action
+      checkForDuplicateTitle: async (
+        newTitle: string,
+        isTitleDuplicateConfirmed: boolean,
+        onTitleDuplicate: () => void
+      ) => {
+        await checkForDuplicateTitle(
+          { title: newTitle, lastSavedTitle: '' },
+          false,
+          isTitleDuplicateConfirmed,
+          onTitleDuplicate
+        );
+      },
```

Comment:
Yes this **was** being handled in the `saveToLibrary` method but it was problematic.

The `dashboard` plugin only passed the `onTitleDuplicate` callback to the `embeddable.checkForDuplicateTitle`.

https://github.com/elastic/kibana/blob/94c0008ea6f29c60fee0013447c87d48244ee9b7/src/platform/plugins/shared/dashboard/public/dashboard_actions/library_add_action.tsx#L94-L100

Because of this the vis plugin would have to use the fallback confirm modal when a duplicate title is found. Even then this would break and just be left in an infinite loading loop.

But this duplicate check should be called when saving from the vis app itself. This now handled the dup check at the dashboard level and only check the duplicate title on save when `onTitleDuplicate` is defined.